### DISCUSSION TITLE
Solax X3-MIC ?

### DISCUSSION THREAD

**Discussion by beocycris** [syssi/esphome-modbus-solax-x1#21] [Q&A]
Does anybody know, if this module also works with the new Solax X3-MIC (G2) ?

regards 
Frank

**Comment by syssi** [comment]
I've found an implementation using Modbus RTU to communicate with the Solax X3:

https://github.com/InfernoEmbedded/PowerScraper/blob/master/Inputs/SolaxX3RS485.py

This component implements another protocol. I'm pretty sure you can use this ESPHome component to talk to your device:

https://esphome.io/components/modbus_controller.html

**Reply by syssi** [reply]
Ping me if you need help. I would love to add a tested example configuration for your solax model to the project.

**Reply by syssi** [reply]
Another idea: May be you have luck and the device supports two protocols.
